### DISCUSSION TITLE
Big angle brackets alternative for <> when used as delimiters

### DISCUSSION THREAD

**Discussion by xAlien95** [tonsky/FiraCode#1217] [Ideas]
One of the downsides most users experience when using `<` and `>` as brackets is what they call "angle bracket blindness". I first faced this expression when discussing about Swift and its generic types.

In brief, "angle brackets" are shorter than the other brackets and the eye may struggle to recognize them as delimiters:
```swift
// easy
struct Optional<Wrapped> { ... }

// not so easy
struct ForEach<
  Data: RandomAccessCollection,
  ID: Hashable,
  Content: View
> { ... }
```

My question is: would it be possible to have full-height alternatives when used as brackets?

```swift
// better
struct Optional〈 Wrapped 〉 { ... }

// better
struct ForEach〈
  Data: RandomAccessCollection,
  ID: Hashable,
  Content: View
〉 { ... }
```

**Comment by tonsky** [comment]
Not when used as brackets, but can be replaced everywhere

**Reply by xAlien95** [reply]
Thank you. So a font can only be context-unaware, is that right? Font ligatures behave like combined emojis?

**Comment by RubenVerg** [comment]
The principle for ligatures is kind of the same, yes. (Your system's emoji font like Segoe UI Emoji on Windows implements ZWJ sequences with ligatures)

- RubenVerg ❤️


________________________________

**Comment by xAlien95** [comment]
Thank you @RubenVerg. So, in order to obtain that behavior in, say, Visual Studio Code, I would need to develop a language server extension that adds a css class to `<` and `>` tokens when used as delimiters.

Was [font-variant-alternates](https://developer.mozilla.org/en-US/docs/Web/CSS/font-variant-alternates) the right way to choose an alternative glyph? It's currently deprecated. How can I set a specific alternate via css?

**Comment by tonsky** [comment]
Yes, font-variant-alternates + styleset()/character-variant() should do the trick, but font has to support this particular replacement.
